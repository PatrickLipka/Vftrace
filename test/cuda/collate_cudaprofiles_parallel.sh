#!/bin/bash
set -x
test_name=collate_cudaprofiles_parallel
exe_name=collate_cudaprofiles_parallel
output_file=${test_name}.out
ref_file=${srcdir}/ref_output/${output_file}

rm -f ${output_file}

nvidia-smi
has_gpu=`echo $?`

${MPI_EXEC} ${MPI_OPTS} ${NP} 2 ./${exe_name} > ${output_file} || exit 1

if [ $has_gpu -eq 0 ]; then
### nvidia-smi has identified GPUs. Remove the corresponding line
  sed -i '/Using [0-9+] GPUs/d' ${output_file}
  sed -i '/Visible GPUs/d' ${output_file}
else
### No GPUs found
   sed -i '/No GPUs available/d' ${output_file}
fi

diff ${output_file} ${ref_file}
