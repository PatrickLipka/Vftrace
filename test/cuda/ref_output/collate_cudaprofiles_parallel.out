Ranklogfile for rank 0: 

--Cuda Summary--


+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
| STID | cudaName  | CBID | #Calls | t_compute[s] | t_memcpy[s] | t_other[s] | Host->Device[B] | Device->Host[B] |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
|    0 | cudafunc4 |   13 |      3 |        30.00 |        0.00 |       0.00 |               0 |               0 |
|    1 | cudafunc3 |   31 |      1 |         0.00 |       20.00 |       0.00 |               0 |          524288 |
|    2 | cudafunc2 |   31 |      1 |         0.00 |        1.00 |       0.00 |         1048576 |               0 |
|    3 | cudafunc1 |   13 |      1 |         1.00 |        0.00 |       0.00 |               0 |               0 |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
Ranklogfile for rank 1: 

--Cuda Summary--


+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
| STID | cudaName  | CBID | #Calls | t_compute[s] | t_memcpy[s] | t_other[s] | Host->Device[B] | Device->Host[B] |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
|    0 | cudafunc4 |   13 |      5 |       100.00 |        0.00 |       0.00 |               0 |               0 |
|    1 | cudafunc3 |   31 |      1 |         0.00 |       40.00 |       0.00 |               0 |          882425 |
|    2 | cudafunc5 |   13 |      1 |        10.00 |        0.00 |       0.00 |               0 |               0 |
|    3 | cudafunc1 |   13 |      1 |         2.00 |        0.00 |       0.00 |               0 |               0 |
|    4 | cudafunc2 |   31 |      1 |         0.00 |        2.00 |       0.00 |         6758401 |               0 |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
Collated logfile: 

--Cuda Summary--


+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
| STID | cudaName  | CBID | #Calls | t_compute[s] | t_memcpy[s] | t_other[s] | Host->Device[B] | Device->Host[B] |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
|    0 | cudafunc4 |   13 |      8 |      130.000 |       0.000 |      0.000 |               0 |               0 |
|    1 | cudafunc3 |   31 |      2 |        0.000 |      60.000 |      0.000 |               0 |         1406713 |
|    2 | cudafunc5 |   13 |      1 |       10.000 |       0.000 |      0.000 |               0 |               0 |
|    3 | cudafunc1 |   13 |      2 |        3.000 |       0.000 |      0.000 |               0 |               0 |
|    4 | cudafunc2 |   31 |      2 |        0.000 |       3.000 |      0.000 |         7806977 |               0 |
+------+-----------+------+--------+--------------+-------------+------------+-----------------+-----------------+
